apps/ide/src/plugins/webida.ide.project-management.run/plugin.js

```diff
@@ -304,8 +305,9 @@ define([
                     return ext.type === projectInfo.type;
                 });
                 if (ext) {
+                    extMethodName = ext.beforeLaunch;
                     require([ext.module], function (mod) {
-                        mod[ext.beforeLaunch](projectInfo, mode, callback);
+                        mod[extMethodName].call(mod, projectInfo, mode, callback);
```

Comment:
(opt) I think 'extFunctionName' or 'extFnName' would be better. And need to check mod[extFnname].